src/nlp_types.jl

```diff
@@ -4,6 +4,27 @@ export has_bounds, bound_constrained, unconstrained, linearly_constrained,
 # Base type for an optimization model.
 abstract type AbstractNLPModel end
 
+show_header(nlp :: AbstractNLPModel) = "$(typeof(nlp))\n"
+
+import Base.show
+function show(io :: IO, nlp :: AbstractNLPModel)
+  output = show_header(nlp)
+  output *= "  $(nlp.meta.nvar) variables "
+  output *= (has_bounds(nlp) ? "with" : "without") * " bounds\n"
+  ncon = nlp.meta.ncon
+  if ncon > 0
+    nequ = length(nlp.meta.jfix)
+    if nequ == 0
+      output *= "  $ncon inequality constraints\n"
+    elseif nequ == nlp.meta.ncon
+      output *= "  $ncon equality constraints\n"
+    else
+      output *= "  $nequ equality constraints and $(ncon - nequ) inequality constraints\n"
+    end
+  end
+  print(io, output)
+end
+
```

Comment:
Let's just show full statistics without trying to be too clever. Something like [this](https://github.com/optimizers/NLPLab/blob/master/%2Bmodel/NlpModel.m#L281) or like what Ipopt and Knitro show before a solve.